MetaMask/metamask-docs
# Unified login

## Changes in this PR:

 - Provide unified login interface with MetaMask login button in the main header
 - Adding a feature flag to hide this button [mm-unified-login](https://app.launchdarkly.com/projects/infura/flags/mm-unified-login/targeting?env=test&env=production&selected-env=test)

## How to test:
 - Make sur MetaMask is not connected and locked
 - Navigate to https://metamask-docs-git-unified-login-metamask-web.vercel.app/wallet/reference/new-reference/ and try to connect with MetaMask
<img width="688" alt="Screenshot 2024-09-19 at 17 36 24" src="https://github.com/user-attachments/assets/9c6e1091-b62e-49d1-8f44-c92370f4a688">

 - You should see a modal like this
<img width="835" alt="Screenshot 2024-09-18 at 18 09 37" src="https://github.com/user-attachments/assets/6c597342-0fd0-4da3-bb9e-e4a73e3393fa">

- If your wallet is linked with an Infura account, nothing to do, you will connect instantaneously

- If your wallet is not linked you will be redirected to the dashboard to enter and link your Infura account with email/password
<img width="835" alt="Screenshot 2024-09-20 at 12 44 40" src="https://github.com/user-attachments/assets/0c4e9b21-0e02-462b-b226-eafd2620e260">

 - If your wallet is linked with multiple Infura account you will be redirected to the dashboard to select which account you want to use
<img width="1480" alt="Screenshot 2024-09-20 at 12 57 17" src="https://github.com/user-attachments/assets/fb12cf5f-2036-4cfa-8c98-ffc9fdb05147">

 - Then you should be redirected to the documentation and see your Infura projects API Keys
<img width="654" alt="Screenshot 2024-09-20 at 12 52 10" src="https://github.com/user-attachments/assets/43824c27-568f-40e1-adad-8ea8f03ef532">

## Links:

[JIRA Issue](https://infura.atlassian.net/browse/ACT-1382)
